libraries/plugins/apis/rewards_api/rewards_api.cpp

```diff
@@ -0,0 +1,80 @@
+#include <chainbase/chainbase.hpp>
+#include <steem/chain/steem_objects.hpp>
+#include <steem/chain/util/reward.hpp>
+#include <steem/chain/util/uint256.hpp>
+#include <steem/plugins/chain/chain_plugin.hpp>
+#include <steem/plugins/rewards_api/rewards_api_plugin.hpp>
+#include <steem/plugins/rewards_api/rewards_api.hpp>
+
+namespace steem { namespace plugins { namespace rewards_api {
+
+namespace detail {
+
+class rewards_api_impl
+{
+public:
+   rewards_api_impl() :
+      _db( appbase::app().get_plugin< steem::plugins::chain::chain_plugin >().db() ) {}
+
+   DECLARE_API_IMPL( (simulate_curve_payouts) );
+
+   chain::database& _db;
+};
+
+DEFINE_API_IMPL( rewards_api_impl, simulate_curve_payouts )
+{
+   simulate_curve_payouts_return ret;
+
+   const auto& cidx = _db.get_index< chain::comment_index, chain::by_cashout_time >();
+
+   auto current = cidx.begin();
+
+   fc::uint128_t sum_current_curve_vshares = 0;
+   fc::uint128_t sum_simulated_vshares = 0;
+
+   std::vector< simulate_curve_payouts_element > elements;
+   std::vector< fc::uint128_t >                  element_vshares;
+
+   auto reward_fund_object = _db.get< chain::reward_fund_object, chain::by_name >( STEEM_POST_REWARD_FUND_NAME );
+
+   while( current != cidx.end() && current->cashout_time < fc::time_point_sec::maximum() )
+   {
+      simulate_curve_payouts_element e;
+      e.author = current->author;
+      e.permlink = current->permlink;
+
+      auto new_curve_vshares = chain::util::evaluate_reward_curve( current->net_rshares.value, args.curve, args.var1 );
+      sum_simulated_vshares = sum_simulated_vshares + new_curve_vshares.to_uint64();
+
+      auto current_curve_vshares = chain::util::evaluate_reward_curve( current->net_rshares.value, reward_fund_object.author_reward_curve, reward_fund_object.content_constant );
+      sum_current_curve_vshares = sum_current_curve_vshares + current_curve_vshares.to_uint64();
+
+      elements.push_back( std::move( e ) );
+      element_vshares.push_back( new_curve_vshares );
+   }
+
+   auto current_estimated_recent_claims = reward_fund_object.recent_claims + sum_current_curve_vshares;
+
+   auto simulated_recent_claims = ( current_estimated_recent_claims / sum_current_curve_vshares ) * sum_simulated_vshares;
+
+   for ( std::size_t i = 0; i < elements.size(); ++i )
+      elements[ i ].payout = protocol::asset( ( element_vshares[ i ] / simulated_recent_claims ).to_uint64(), STEEM_SYMBOL ) * reward_fund_object.reward_balance;
+
+   ret.payouts = std::move( elements );
```

Comment:
Not that I am too concerned with efficiency here, but I would just build that payouts list in the return type rather than creating an extra temporary variable. I wouldn't be surprised if the compiler optimizes to this.